crates/bevy_ecs/src/event.rs

```diff
@@ -34,10 +36,12 @@ pub trait Event: Send + Sync + 'static {}
 /// sent to the point it was processed. `EventId`s increase montonically by send order.
 ///
 /// [`World`]: crate::world::World
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
 pub struct EventId<E: Event> {
     /// Uniquely identifies the event associated with this ID.
     // This value corresponds to the order in which each event was added to the world.
     pub id: usize,
+    #[reflect(ignore)]
```

Comment:
This needs to be wrapped in a `#[cfg_attr]`.

Reply:
![image](https://github.com/bevyengine/bevy/assets/53269692/6908f174-e202-4961-b913-3a44bac14e99)